src/utils/sql.ts

```diff
@@ -1,23 +1,32 @@
 import * as pg from "pg";
 import { QueryFragment, QueryFragmentBuilder } from "../types";
 
+function* indexFactory(seed = 0) {
```

Comment:
Maybe call this `sequence(Factory|Generator)?` instead. `index` is really such a meaningless word (in this context).